sonar-scala-plugin/src/main/scala/org/sonarsource/scala/converter/ScalaConverter.scala

```diff
@@ -221,8 +223,17 @@ class ScalaConverter extends ASTConverter {
       }
     }
 
-    def grandParentIsNewAnonymous(variable: Stat) = {
-      variable.parent.exists(p => p.is[Template] && p.parent.exists(gp => gp.is[Term.NewAnonymous]))
+    def greatGrandParentIsNewAnonymous(variable: Stat) = {
+      variable.parent.exists(p => {
+        val isTempBdy = p.is[Template.Body];
+        val grandParentIsTemplate = p.parent.exists({
+          gp => gp.is[Template]
+        })
+
+        val greatGreatParents = p.parent.flatMap(gp => gp.parent)
```

Comment:
Can there be more than 1 grand parent? It looks to me like the `.parent` api only returns a single parent or nothing. In which case we can simply rename the variable here